### DISCUSSION TITLE
Idea for GSoC 2023 - Graph edge coloring

### DISCUSSION THREAD

**Discussion by mam15** [networkx/networkx#6549] [Algorithms]
Hello! My name is Matheus, and I'd like to propose a new idea for GSoC 2023, related to graph edge coloring. I noticed that the coloring algorithms available at NetworkX are only related to vertex coloring, and I think the addition of edge coloring algorithms would prove beneficial to users, as edge coloring has many applications and is a widely studied subject. My general idea is to implement two algorithms, one for optimal edge coloring (which is np-hard), and another for Delta+1 coloring (which has polynomial time algorithms); there are several algorithms available for both tasks, but I'll leave this decision open so that it can be discussed with a mentor, should one be interested. I have experience in this subject, as I developed an optimization for SageMath related to edge coloring (available [here](https://github.com/sagemath/sage/issues/34809)), and also wrote a research project related to this algorithm and implementation as a requirement for my Bachelor's degree in CS. I graduated from UFPR last semester, so I have full time availability for GSoC.

Please let me know if any mentors would be interested in this idea, so that I may come up with a more structured document describing my competences and the project for feedback and further discussion. Thank you in advance for your consideration!

**Comment by Kemsekov** [comment]
I actually just use vertex coloring on the line graph of graph and treat verticies as edges ¯\_(ツ)_/¯

**Comment by padath314** [comment]
Is this implementation opportunity still open? Me and @Sreelaxme CSE students eager to contribute to this project. It's an exciting initiative, and we believe adding edge coloring algorithms to NetworkX would be a valuable enhancement. Could you please provide us with guidance on how to proceed?

**Reply by Kemsekov** [reply]
Look at what line graph is.

It is just converting graph edges into vertices that is connected if edges have common vertex.

You get edge colouring for free if you just apply vertex colouring on the line graph.
